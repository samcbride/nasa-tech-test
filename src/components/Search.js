import React, { useState } from "react";
import "../styles/Search.css";

const Search = () => {
  return (
    <>
      <input className="search-area" type="text"></input>
    </>
  );
};

export default Search;
