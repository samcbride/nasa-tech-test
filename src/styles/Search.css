form {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.search-area {
  width: 25%;
  height: 20px;
  font-size: 20px;
  border-radius: 10px;
  border-color: #212529;
  margin: 0em 0.5em;
}

.search-button {
  border-radius: 10px;
  padding: 0.15em 0.75em;
  background-color: coral;
  border-color: #212529;
  font-family: "Nanum Gothic", sans-serif;
  color: aliceblue;
}
