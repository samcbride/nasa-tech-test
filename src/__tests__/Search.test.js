import React from "react";
import { render } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  const { asFragment } = render(<Search />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("check button exists with correct text", () => {
    const { getByRole } = render(<Search />);
    const button = getByRole("button", { name: /Go/i });
    expect(button).toBeInTheDocument();
  });
});
