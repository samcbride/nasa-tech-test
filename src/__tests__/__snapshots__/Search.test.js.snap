// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <form
      class="search-form"
    >
      <input
        class="search-area"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
